include/vara/Feature/Feature.h

```diff
@@ -50,18 +50,6 @@ class Feature : public FeatureTreeNode {
 
   [[nodiscard]] bool isOptional() const { return Opt; }
 
-  /// Search parent feature in tree structure -- this may not exist or nullptr
-  /// if node is already root.
-  [[nodiscard]] Feature *getParentFeature() const {
```

Comment:
Why did you move this to the `TreeNodeClass`? As an abstraction this makes more sense on a Feature, with the move the TreeNode is now hard Tied to the Feature.

Reply:
`Relationship` now needs a look up to a parent feature too. I reworked this as a template in `FeatureTreeNode`.

Reply:
I fixed the Relationship with local cast handling, we don't really need the template.